### ISSUE TITLE
[java] ReturnEmptyCollectionRatherThanNull should disallow potential null return 

### ISSUE THREAD

**Issue by MetaBF** [pmd/pmd#3880]
**Affects PMD Version:**
6.44.0+
7.0.0

**Rule:** [ReturnEmptyCollectionRatherThanNull](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#returnemptycollectionratherthannull)

**Description:**
Currently, the rule **ReturnEmptyCollectionRatherThanNull** detects `return null` statements, which I think is not enough, because it is not the only way that a function can produce a null value. In the following example, function `get()` will return `null` (if `flag` is false),  I think it should be warned. 

**Code Sample demonstrating the issue:**

```java
import java.util.ArrayList;
import java.util.List;

public class ExampleReturnEmptyArrayRatherThanNull {

    public boolean flag;
    public List<String> get(){
        List<String> res = null; // line 8
        if(flag) { 
            res = new ArrayList<>();
            res.add("");
        }
        return res; // line 13, false-negative
    }
}

```

**Expected outcome:**

PMD should report a violation at line 13, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
